narwhals/_arrow/dataframe.py

```diff
@@ -49,7 +48,11 @@ def __narwhals_namespace__(self) -> ArrowNamespace:
         return ArrowNamespace(backend_version=self._backend_version)
 
     def __native_namespace__(self) -> Any:
-        return get_pyarrow()
+        if self._implementation is Implementation.PYARROW:
+            return self._implementation.to_native_namespace()
+
+        msg = f"Expected pyarrow, got: {type(self._implementation)}"  # pragma: no cover
+        raise AssertionError(msg)
```

Comment:
Do we need this check? There was in pandas, so I added it everywhere

Reply:
probably not needed (but doesn't hurt to have it)